### PR TITLE
AzureRT S54 PR #3 - Extension Image List

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@huangpf__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (phuang). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: test/commands/arm/vm/arm.vm-image-list-tests.js [hunk 1]
```diff
@@ -92,6 +93,47 @@ describe('arm', function() {
...
+          done();
+        });
+      });
+
+      it('extension list-image-types ', function(done) {
+        var cmd = util.format('vm extension list-image-types %s %s --json', location, publisher).split(' ');
+        testUtils.executeCommand(suite, retry, cmd, function(result) {
+          result.exitStatus.should.equal(0);
+          var allResources = JSON.parse(result.text);
+          type = allResources[0].name;
```
Comment (yugangw-msft) [inline test/commands/arm/vm/arm.vm-image-list-tests.js]: do you need to check the value?

File: test/commands/arm/vm/arm.vm-image-list-tests.js [hunk 2]
```diff
@@ -92,6 +93,47 @@ describe('arm', function() {
...
+          done();
+        });
+      });
+
+      it('extension list-image-versions ', function(done) {
+        var cmd = util.format('vm extension list-image-versions %s %s %s --json', location, publisher, type).split(' ');
+        testUtils.executeCommand(suite, retry, cmd, function(result) {
+          result.exitStatus.should.equal(0);
+          var allResources = JSON.parse(result.text);
+          version = allResources[0].name;
```
Comment (yugangw-msft) [inline test/commands/arm/vm/arm.vm-image-list-tests.js]: same here.

File: test/commands/arm/vm/arm.vm-image-list-tests.js (hunk 1 above)
Comment (yugangw-msft) [inline test/commands/arm/vm/arm.vm-image-list-tests.js]: also, you will need to declare the variable "var type;", so it won't go to the global scope.

File: test/testlistarmvm.txt [hunk 3]
```diff
@@ -1,6 +1 @@
-commands/arm/vm/arm.vm-create-tests.js
```
Comment (yugangw-msft) [inline test/testlistarmvm.txt]: do you mean to remove all those tests?

Comment (huangpf) [inline test/testlistarmvm.txt]: We reduce this to one or several tests that will be run for runners. New tests shall still be added to ```testlistarm.txt```.

File: test/commands/arm/vm/arm.vm-image-list-tests.js (hunk 1 above)
Comment (huangpf) [inline test/commands/arm/vm/arm.vm-image-list-tests.js]: Actually, it's meant to be global: ```var location, publisher, offer, sku, type, version;``` at line 29.

File: test/commands/arm/vm/arm.vm-image-list-tests.js (hunk 2 above)
Comment (huangpf) [inline test/commands/arm/vm/arm.vm-image-list-tests.js]: Actually, it's meant to be global: ```var location, publisher, offer, sku, type, version;``` at line 29.

Comment (huangpf) [inline test/commands/arm/vm/arm.vm-image-list-tests.js]: The workflow is to use the return fields of test ```i``` as for input for test ```i + 1```, so as long as they don't run in parallel, they should pass in sequence.

**Pull request merged**